app/models/batch_form.rb

```diff
@@ -32,7 +32,6 @@ def self.human_attribute_name(*args)
   delegate :id, :new_record?, :persisted?, to: :batch
 
   validates_presence_of :date_produced
-  validates_numericality_of :samples_quantity, greater_than: 0, message: "value must be greater than 0", if: :creating_batch?
```

Comment:
We should probably make sure the value is non-negative by keeping the validation with `greater_than_or_equal_to: 0` (and appropriate change to the message).

Reply:
I could keep the validation but for what I show on the second screenshot this line of code will be never reached... Should we then modify `views/batches/_form.haml` removing the min in the input??
The "drawback" that I see on this is that pressing the down arrow when value is zero goes to -1 and before the limit was zero, which is a better behaviour.

Reply:
> I could keep the validation but for what I show on the second screenshot this line of code will be never reached... 

Probably not, but it's still important to validate the data we receive in the backend. Client request data can't really be trusted, ever.
There's probably not much practical relevance in this case, but it's a general good practice to validate incoming data.

> Should we then modify views/batches/_form.haml removing the min in the input??

No, the immediate feedback from the HTML validation is good.

We should have both. The roles are distributed:

* Server-side validation is necessary to ensure data integrity.
* Client-side validation is an enhancement on top of that to improve UX

Reply:
TLDR:

- Server side validation -> required for security & data integrity, we must NEVER trust any client provided data, requests can (and will) be forged.

- Client side validation -> improves the user experience with a reactive UI that gives quick feedback, and also prevents useless requests to the backend. When the form is submitted it should be a happy path, with only some backend specific validations that may fail.